sdk/identity/Azure.Identity/tests/ChainedTokenCredentialLiveTests.cs

```diff
@@ -1,260 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Why remove and not disable?

Reply:
see my updated  issue description above.

Reply:
I don't think ChainedTokenCredentialTests and ChainedTokenCredentialLiveTests cover the same thing. If you look at the PR where it was added ChainedTokenCredentialLiveTests seems to cover exception handling and diagnostic scope logic.

Reply:
I would rather fix these tests than remove them.

Reply:
I think it is all covered elsewhere. Each credential has its own `*CredentialLiveTests` which tests exception handling and scope logic (via `InstrumentClient`). 

The `ChainedTokenCredentialLiveTests` only tested chained credential flow with the actual credential implementations, rather than mock implementations. However the mock implementation tests already exercise the only distinct functionality of `ChainedTokenCredential`.

Reply:
@pakrym I agree with @crhistothes that testing the `ChainedTokenCredential` with live tests doesn't make much sense, as the `ChainedTokenCredential` doesn't do any network calls of it's own. These tests were using other credentials which have live test or recorded tests of their own. Testing the functionality of `ChainedTokenCredential` using these credentials which are half live and half mocked really just obscures what we're trying to test and leads to the intermittent failures we're seeing which aren't directly related to the `ChainedTokenCredential`.

Reply:
Yeah, I wish we knew what the test failure is related to. I looked through the code and it looks really straightforward, not something I would expect to be this flaky.

Reply:
I'm OK if we don't want to spend time on this.

Reply:
I also think they bring some coverage. Try commenting out https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/src/ChainedTokenCredential.cs#L87 this like and see if any other test would fail.

Reply:
These fail if I comment that out:

![image](https://user-images.githubusercontent.com/1279263/107099852-6761ed80-67d8-11eb-875c-60bcd72e88c9.png)

Reply:
No idea why it didn't show in a local run :) I give up :)

Reply:
No worries :)

![](https://media.giphy.com/media/YoKg8cNLtfLulCPEKg/giphy.gif)

Reply:
> Yeah, I wish we knew what the test failure is related to. I looked through the code and it looks really straightforward, not something I would expect to be this flaky.

I think we still have a deadlocking issue either in the `AzureCliCredential` or the `ProcessRunner` we should increase coverage there to see if we can replicate this failure directly with those.

Reply:
It's also possible that there's an issue with how these tests were mocking sub process responses, since both `VisualStudioCredential` and `AzureCLICredential` require this.

Reply:
> It's also possible that there's an issue with how these tests were mocking sub process responses, since both `VisualStudioCredential` and `AzureCLICredential` require this.

That's my guess as well.